mobot/mobot_client/timeouts.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2021 MobileCoin. All rights reserved.
+
+import datetime
+import pytz
+import time
+
+from mobot_client.models import DropSession, Customer, Message
+from mobot_client.drop_session import ItemSessionState
+from mobot_client.chat_strings import ChatStrings
+
+utc = pytz.UTC
+
+
+class Timeouts:
+
+    def __init__(self, messenger, schedule=30, idle_timeout=60, cancel_timeout=300):
+        self.messenger = messenger
+        self.schedule = schedule
+
+        # Time before warning
+        self.idle_timeout = idle_timeout
+
+        # Time before cancelling session and refunding payment
+        self.cancel_timeout = cancel_timeout
+
+    @staticmethod
+    def customer_is_active(customer):
+        for session in DropSession.objects.filter(customer=customer).values('state'):
+            if session['state'] in ItemSessionState.active_states():
+                return True
+        return False
+
+    @staticmethod
+    def customer_is_idle(customer):
+        for session in DropSession.objects.filter(customer=customer).values('state'):
+            if session['state'] == ItemSessionState.IDLE.value:
+                return True
+        return False
+
+    @staticmethod
+    def customer_needs_refund(customer):
+        for session in DropSession.objects.filter(customer=customer).values('state'):
+            if session['state'] in ItemSessionState.refundable_states():
+                return True
+        return False
+
+    @staticmethod
+    def set_customer_idle(customer):
+        for session in DropSession.objects.filter(customer=customer):
+            if session.state in ItemSessionState.active_states():
+                session.state = ItemSessionState.IDLE.value
+                session.save()
```

Comment:
This can be replaced with something like
`DropSession.objects.filter(customer=customer, state__in=ItemSessionState.active_states()).update(state=ItemSessionState.IDLE.value)`

However, it's pretty funky to not have this operate on a specific session. If there is a bug and we somehow ended with multiple active sessions, we will not find out and this code would just manipulate both of them.

It appears to me that perhaps a `Customer` should have a nullable `active_session` field.